pkg/util/startup/startup.go

```diff
@@ -0,0 +1,38 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package startup
+
+// AutoStart defines whether a component or bundle should start automatically.
+type AutoStart int
+
+const (
+	// Never means that the component should never start.  This is the zero value.
+	Never AutoStart = 0
+
+	// Always means that the component should always start.
+	Always AutoStart = 1
+
+	// IfConfigured means that the component should consult its configuration
+	// (e.g., `foo-agent.enabled`) to decide whether to start.  Components which
+	// have no configuration treat this as Always
+	IfConfigured = 2
+)
+
+// ShouldStart helps a component determine whether it should start; "enabled"
+// is the component's configuration value, or `true` for components that are
+// not controlled by a configuration parameter.
+func (a AutoStart) ShouldStart(enabled bool) bool {
```

Comment:
I included "Never" because I'm concerned that dependencies may quietly, unexpectedly end up instantiating a component.  For example, perhaps a component comp/dogstatsd/stats in the dogstatsd bundle provides a flare.Registration, meaning that comp/core/flare depends on comp/dogstatsd/stats.  Now a relatively innocuous addition of a dependency on `comp/core/flare` would instantiate dogstatsd, and if there's no "Never" option then it will start and listen on 8126 and cause mayhem. 

I do agree that IfConfigured should be the default value.